qiskit/synthesis/evolution/product_formula.py

```diff
@@ -25,7 +25,7 @@
 class ProductFormula(EvolutionSynthesis):
     """Product formula base class for the decomposition of non-commuting operator exponentials.
 
-    Lie-Trotter and Suzuki inherit this class.
+    :obj:`.LieTrotter` and :obj:`.SuzukiTrotter` inherit from this class.
```

Comment:
They affect how Sphinx searches for objects.  The current Sphinx module of this (`ProductFormula`) object is `qiskit.synthesis` I think, so if I just wrote `` `:obj:`LieTrotter` ``, it would try in order:
- `LieTrotter`
- `qiskit.synthesis.LieTrotter`
- `qiskit.synthesis.ProductFormula.LieTrotter`

and if none of those matched, it'd fail.  If you put the dot, it searches those same things (but in reverse order), but if it fails, then it searches for _any_ suffix match, so it'd also try things like `qiskit.quantum_info.LieTrotter`, which otherwise wouldn't be in the path at all.  In this particular case, either would work, but I typically use the `.` to trigger the full suffix match - the way `automodule` and our docstrings manage the `currentmodule` is very confusing (hence we have so many broken cross-references even when they try to use the "full" form), and it's much easier just to force a search if you know there's only one valid object.

Reply:
Ah ok, makes sense -- thanks!

Reply:
I'm not sure _anything_ about Sphinx' process makes sense, to be honest, but I'm fairly sure this is the way things are!